main.go

```diff
@@ -404,7 +406,10 @@ func main() {
 
 		}
 
-		if err = otelv1beta1.SetupCollectorWebhook(mgr, cfg, reviewer, crdMetrics); err != nil {
+		reconciler := controllers.NewReconciler(controllers.Params{})
+		bv := otelv1beta1.SetupCollectorWebhook(mgr, cfg, reviewer, crdMetrics, reconciler.Validate)
+
+		if err = bv; err != nil {
 			setupLog.Error(err, "unable to create webhook", "webhook", "OpenTelemetryCollector")
```

Comment:
Can't we create the reconciler inside the `SetupCollectorWebhook` method? Because I think we aren ot using anything from this scope to create the `reconciler` object, right?